maestro-agent/src/main/java/org/maestro/agent/base/MaestroAgent.java

```diff
@@ -18,98 +21,140 @@
 
     private static final Logger logger = Logger.getLogger(MaestroAgent.class);
     private GroovyHandler groovyHandler;
+    private AbstractConfiguration config = ConfigurationWrapper.getConfig();
+    private File path;
 
 
     public MaestroAgent(String maestroURL, String role, String host, MaestroDataServer dataServer) throws MaestroException {
         super(maestroURL, role, host, dataServer);
+
+        String pathStr = config.getString("maestro.agent.ext.path", null);
+
+        if (pathStr == null){
+            URL uri = this.getClass().getResource("/org/maestro/agent/ext/requests/");
+            if (uri != null) {
+                pathStr = uri.getPath();
+            }
+            else{
+                logger.error("Unable to load files for extension points.");
+                pathStr = Constants.HOME_DIR + File.separator + "ext";
+            }
+        }
+
+        path = new File(pathStr);
+
         groovyHandler = new GroovyHandler(super.getClient());
     }
 
     public void handle(StartInspector note) {
-
+        File entryPointDir = new File(path, "startinspector");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StartReceiver note) {
-
+        File entryPointDir = new File(path, "startreceiver");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StartSender note) {
-
+        File entryPointDir = new File(path, "startsender");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StopInspector note) {
-
+        File entryPointDir = new File(path, "stopinspector");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StopReceiver note) {
-
+        File entryPointDir = new File(path, "stopreceiver");
+        callbacksWrapper(entryPointDir);
     }
 
     public void handle(StopSender note) {
-
+        File entryPointDir = new File(path, "stopsender");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(StatsRequest note) {
         super.handle(note);
+
+        File entryPointDir = new File(path, "stats");
+        callbacksWrapper(entryPointDir);
     }
 
     @Override
     public void handle(FlushRequest note) {
         super.handle(note);
 
-        URL uri = this.getClass().getResource("/org/maestro/agent/ext/requests/flush/");
-        if (uri != null) {
-            String path = uri.getPath();
-            File file = new File(path);
-            try {
-                groovyHandler.setInitialPath(file);
-                groovyHandler.runCallbacks();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }
-        else{
-            logger.error("Unable to load files for extension points.");
-        }
-
-
+        File entryPointDir = new File(path, "flush");
```

Comment:
@Frawless Let's move the strings with the directory names to String constants. IMHO, that makes refactoring easy if needed.